core/cache/application.go

```diff
@@ -76,6 +76,16 @@ type appCharmUrlChange struct {
 func (a *Application) setDetails(details ApplicationChange) {
 	a.mu.Lock()
 
+	// If this is the first receipt of details, set the removal message.
+	if a.details.Name == "" {
```

Comment:
Why not make `a.removalMessage` a pointer, then you can do a nil check. It removes the computational load of trying to work out if the name of the details can ever be empty?

Reply:
Good idea.